design/one-pager-function-go-templating.md

```diff
@@ -0,0 +1,171 @@
+# Go Templating Function
+ 
+* Owner: Ezgi Demirel (@ezgidemirel)
+* Reviewers: Crossplane Maintainers
+* Status: Accepted
+
+## Background
+
+With Crossplane 1.14, Composition Functions will be promoted to `v1beta1`, and several 
+composition functions will be introduced as an extension point to be used in pipelines. 
+One of these extensions is the [Go Templating Function], which allows users to render Crossplane
+resources using Go templating capabilities like conditionals and loops. In addition to that, 
+users can use values in the environment configs or other resource fields to create/configure
+Crossplane resources.
+
+## Getting Template Files
+
+With the initial implementation of the Go Templating Function, users will be able
+to provide templates in two different ways:
+- An `inline` source can be used to provide template files as a string in the
+  `input.inline.template` field of the pipeline step.
+- A `fileSystem` source can be used to provide template files with folder path
+  in the `input.fileSystem.dirPath` field of the pipeline step.
+
+Both of these options have their own advantages and disadvantages. While `inline` source
+is faster and easier to use, it is not suitable for large template files and hard to maintain.
+On the other hand, `fileSystem` source requires the files to be present in the local file system,
+which might be possible by mounting config maps with `DeploymentRuntimeConfig` or building another
+container image on top of the existing one with the template files. 
+
+In the next iterations, it is planned to support fetching template files from a remote location like
+GitHub repositories or S3 buckets.
```

Comment:
This is fascinating. Would pushing to git repository (approving PR) enable GitOps scenarios where the changed template would be fetched by the running composition and updated in the system by creating a new composition revision? There are some pros/cons to this.

Reply:
We can implement something like `templatePullPolicy`. But as you mentioned, providing a gitOps tooling has pros and cons.